core/tutorialbar.py

```diff
@@ -52,14 +56,23 @@ def get_course_links(self, url: str) -> List:
         :return: list of pages on tutorialbar.com that contain Udemy coupons
         """
         response = requests.get(url=url)
+
         soup = BeautifulSoup(response.content, "html.parser")
+
         links = soup.find("div", class_="rh-post-wrapper").find_all("a")
-        self.last_page = links[-2].text
+        # print(links)
+
         courses = []
 
         x = 0
-        for _ in range(self.links_per_page):
-            courses.append(links[x].get("href"))
+        for _ in range(12):
```

Comment:
In the ticket I mentioned these were the Udemy categories. 
This is the tutorialbar logic. 
I don't think they follow the same structure as Udemy where there are clearly defined categories and sub-categories. 
It will still work at this level but will probably not filter out as much as the user would hope

Reply:
Okay I have fixed it by checking whether the category _or_ the subcategory on the **Udemy** page is listed in the **user's** category list.